.github/actions/generate-attestations/action.yml

```diff
@@ -0,0 +1,22 @@
+name: 'Generate Attestations'
+description: 'Generates in-toto attestations based on a SLSA output layout and a predicate'
+inputs:
+  slsa-outputs-file:
+    description: 'A JSON file describing the SLSA output layout with attestation filename keys and the generated subjects (and digests)'
+    required: true
+  predicate-type:
+    description: 'A URI dedining the type of the predicate, for e.g. https://slsa.dev/provenance/v0.2'
+    required: true
+  predicate-file:
+    description: 'A JSON file describing the SLSA predicate to attach to the subjects'
+    required: true
+  output-folder:
+    description: 'Output folder to place attestations'
+    required: false
+    default: 'attestations'
```

Comment:
how will this work if the Action is called several times, e.g. if there are multiple `builder_labels` running in parallel or if the PW has calls the TRW multiple times in the same workflow? The name needs to be unique. Would it be better to remove this input and only provide the output, which this Action would randomize `attestations-xxxxxx`?
Maybe this will be done by the SRW which we control and we will re-use the rng Action for it? If that's the case, shall we mark the input as `required`, since the default is not safe?

Reply:
> Maybe this will be done by the SRW which we control and we will re-use the rng Action for it? If that's the case, shall we mark the input as required, since the default is not safe?

Yes, this will be done by the SRW. Something like here during the trusted attestation creation/signing: https://github.com/laurentsimon/slsa-delegator/blob/main/.github/workflows/delegator_generic_slsa3.yml#L237-L244

Ah, sg, let's mark it as required. I agree it was confusing to have a default anyway. Having a required input also means that I no longer need to set an output.

Reply:
SG.